go/objecthash/objecthash.go

```diff
@@ -171,25 +176,59 @@ func hashBool(b bool) [hashLength]byte {
 	return hash(`b`, bb)
 }
 
+func hashBytes(value reflect.Value) ([hashLength]byte, error) {
+	var b []byte
+	if value.Kind() == reflect.Slice {
+		b = value.Bytes()
+	} else {
```

Comment:
Consider doing a second type check here to catch calls of `hashBytes` with a non `[]byte` value passed in.

Reply:
See #41